### PR TITLE
Outer filters hidden when column hidden

### PR DESCRIPTION
**Pull request by jakubvojacek**
I do not think that outer filters should be displayed when curespondive column is hidden.

### CODE REVIEW

**Comment by dakorpar** [comment]
This could crash existing grids, for example I have grids where you have filters defined for something that's not displayed at all, and sometimes user still wants to filter something although they don't need to see it in grid.

**Comment by jakubvojacek** [comment]
Well filters are hidden they're not outer and the column is hidden - I would expect that same behavior for outer filters. However I will add method to this PR, something like `Column::hideFilterWhenColumnHidden(TRUE|FALSE)`

**Comment by dakorpar** [comment]
@jakubvojacek @paveljanda  This broke down existing code in my project, probably in some elses to, Default here should be like previous behaviour, and allowing users to hide it itself, plus It would be nice to have this on whole grid for all columns?

File: src/templates/datagrid.latte [hunk 1]
```diff
@@ -26,25 +26,29 @@
...
-						{var $filter_type_block = 'filtertype-' . $f->getType()}
-
-						{ifset #$filter_block}
-							{include #$filter_block, filter => $f, input => $form['filter'][$f->getKey()], outer => TRUE}
-						{else}
-							{ifset #$filter_type_block}
-								{include #$filter_type_block, filter => $f, input => $form['filter'][$f->getKey()], outer => TRUE}
-								{else}
-									{include $f->getTemplate(), filter => $f, input => $form['filter'][$f->getKey()], outer => TRUE}
+						{continueIf !array_key_exists($f->getKey(), $columns)}
```
Comment (juniwalk) [inline src/templates/datagrid.latte]: Hello, could you use `!isset($columns[$f->getKey()]`? It is faster and should not cause any issues in this case. Also you could possibly get filter key from the foreach above.

**Comment by juniwalk** [comment]
Your idea about adding `Column::hideFilterWhenColumnHidden(TRUE|FALSE)` sounds great! Could you implement it?

**Comment by dakorpar** [comment]
Why not both global and on column itself?

**Comment by juniwalk** [comment]
@dakorpar Well no reason, I just tend to keep things simple. This option is of course possible too.

**Comment by paveljanda** [comment]
@dakorpar Any further intentions to continue with this PR? :)

**Comment by dakorpar** [comment]
@paveljanda that's not my PR :)

**Comment by paveljanda** [comment]
@dakorpar Oh, sorry

@jakubvojacek Any further intentions to continue with this PR? :)

**Pull request closed without merging**